src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java

```diff
@@ -93,17 +95,25 @@
             Object[] selfArgs = new Object[args.length + 1];
             selfArgs[0] = receiver;
             System.arraycopy(args, 0, selfArgs, 1, args.length);
+            Method foundDgmMethod = null;
+
             for (Class<?> dgmClass : DGM_CLASSES) {
                 Method dgmMethod = GroovyCallSiteSelector.staticMethod(dgmClass, method, selfArgs);
                 if (dgmMethod != null) {
                     if (whitelist.permitsStaticMethod(dgmMethod, selfArgs)) {
                         return super.onMethodCall(invoker, receiver, method, args);
-                    } else {
-                        throw StaticWhitelist.rejectStaticMethod(dgmMethod);
+                    } else if (foundDgmMethod == null) {
+                        foundDgmMethod = dgmMethod;
                     }
                 }
             }
 
+            // Some methods are found by GroovyCallSiteSelector in both DefaultGroovyMethods and StringGroovyMethods, so
+            // we're iterating over the whole list before we decide to fail out on the first failure we found.
```

Comment:
Makes sense, I think, but would be nice to have a test case proving that this change is effective.